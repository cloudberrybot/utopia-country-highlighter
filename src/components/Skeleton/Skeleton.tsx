
interface ISkeletonProps {
  columns: number
  rows?: number
  flow: 'col' | 'row'
}

function Skeleton({ flow, columns, rows }: ISkeletonProps) {
  if (flow === 'col') {
    return <>{Array.from(Array(columns).keys()).map(value => (<div key={value} className="w-full h-8 animate-pulse bg-gray-400"></div>))}</>
  }
  return <div className="w-40 animate-pulse bg-gray-400 py-2 px-2"></div>;
}

export default Skeleton;