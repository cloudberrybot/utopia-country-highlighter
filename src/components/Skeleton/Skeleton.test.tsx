import '@testing-library/jest-dom'

import Skeleton from './Skeleton';
import { render } from '@testing-library/react';

test('renders correctly', () => {
  const { asFragment } = render(
    <Skeleton flow="col" columns={5} />
  );

  expect(asFragment()).toMatchSnapshot();
});
