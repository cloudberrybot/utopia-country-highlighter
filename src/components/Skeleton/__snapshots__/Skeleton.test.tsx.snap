// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full h-8 animate-pulse bg-gray-400"
  />
  <div
    class="w-full h-8 animate-pulse bg-gray-400"
  />
  <div
    class="w-full h-8 animate-pulse bg-gray-400"
  />
  <div
    class="w-full h-8 animate-pulse bg-gray-400"
  />
  <div
    class="w-full h-8 animate-pulse bg-gray-400"
  />
</DocumentFragment>
`;
